client/chat/action-creators.ts

```diff
@@ -270,34 +271,32 @@ export function retrieveUserList(channelId: SbChannelId): ThunkAction {
   }
 }
 
-const chatUserProfileLoadsInProgress = new Set<`${SbChannelId}|${SbUserId}`>()
-
 export function getChatUserProfile(
   channelId: SbChannelId,
   targetId: SbUserId,
   spec: RequestHandlingSpec<void>,
 ): ThunkAction {
   return abortableThunk(spec, async (dispatch, getStore) => {
+    const requestCoalescer = new RequestCoalescer()
```

Comment:
This doesn't work, you're making a new coalescer for every request so it will coalesce nothing. It needs to be global

Reply:
Ah, that's a silly mistake.

You know, when I first started implementing this, I actually didn't even use a class. I just wrote a function and had the in progress requests saved in a global Map. But then I realized that it would be hard to ensure that the IDs we use are always unique for different APIs. So I deleted it and went with the class approach.

But maybe using a function is not a bad idea after all? And for the in progress Map keys, we just use whatever we call in `apiUrl`, e.g. `users/${userId}/profile`, `chat/${channelId}/users/${targetId}`, and `games/${gameId}`.

Reply:
No, I think you should use the class as you have an create a global instance for each particular action that needs to be coalesced